src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -180,9 +180,12 @@ internal IList<string> ReadStringArrayAsIList(IList<string> strings = null)
                 {
                     string value = _reader.ReadTokenAsString();
 
-                    strings ??= new List<string>();
+                    if (value != null)
```

Comment:
Would it be "better" to throw, rather than silently ignore what we consider a badly formed assets file?  On one hand, silently ignoring nulls will not fail builds, at least not immediately, and not in a code path that NuGet will get blamed for.  On the other hand, if we think that this should never happen, by silently ignoring it, customers are likely to get obscure build errors which will be harder to track down.  If the problem is that the assets file has a null where it shouldn't, the earlier we raise an error about it, the quicker we'll figure out the root cause.

Reply:
The concern is that we just don't know. 
Unfortunately, our reading implementation is all over the place, with it sometimes being permissive and other times not, and this is one of those where I'm not sure I want to rock the boat too much until it proves to be a bigger problem. 

That was part of the concern in the PR body.

Reply:
I'm ok with this not happening and us tabling the idea of making the parsing stricter in a future change.
With the 2 reading implementations and with both of them being inconsistent, I feel like we might be better to wait for 1 to go away before we remove the null safety checks.